reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -61,6 +62,28 @@ static Context of(Object key, Object value) {
 		return new Context1(key, value);
 	}
 
+	/**
+	 * Checks for duplicate keys. This method is intended for a short space of keys in the
+	 * 4-10 range. Shorter number of keys can easily be checked with direct equals comparison(s),
+	 * saving on allocating a vararg array (although this method would still behave correctly).
+	 *
+	 * @param keys the keys to check for duplicates, by looping over the combinations
+	 */
+	static void checkDuplicateKeys(Object... keys) {
```

Comment:
I just realized.... Since the method is defined in the interface, it will be public. I guess we don't want to leak it to public, don't we?

Reply:
ah snap you're right we don't want that to leak... since the approach here is to throw, we don't have to decide between several candidate implementations so we could move this logic in the constructors 🤔

Reply:
you could also create an internal class in `Context.java` just for the validation

Reply:
I moved the static method to `Context4`, which is accessible by `Context5` (the only other usage of the method)